components/hydrator/internal/certsubjectmapping/loader.go

```diff
@@ -160,8 +161,18 @@ func (cl *certSubjectMappingLoader) loadCertSubjectMappings(ctx context.Context,
 		if err != nil {
 			return mappings, errors.Wrap(err, "while listing certificate subject mappings from DB")
 		}
+
+		// The graphql response could be nil, and there could be NO graphql error in case something else failed.
+		// E.g., an error from the oathkeeper or other components
+		if csmGQLPage == nil {
+			return mappings, errors.Errorf("the certificate subject mappings page cannot be nil")
+		}
 		csmTotalCount = csmGQLPage.TotalCount
 		mappings = append(mappings, convertGQLCertSubjectMappings(csmGQLPage.Data)...)
+
+		if csmGQLPage.PageInfo == nil {
+			return mappings, errors.Errorf("the certificate subject mappings page info be nil")
```

Comment:
the certificate subject mappings page info cannot be nil

Reply:
It could in case of non-graphql error(e.g. ory). We even encounter that issue causing the hydrator component to panic

Reply:
I meant that the message should be `the certificate subject mappings page info cannot be nil`. The word `cannot` is missed in the current version

Reply:
Oh, my bad. I thought the comment was about the `if` statement. I fix the error message.